open-apparel-registry/open-apparel-registry
# Enable deleting users

## Overview

- enable users to be deleted through the Django admin by adding a
missing `allauth.socialaccount` app to installed apps
- update migration to make Contributor a nullable field on FacilityLists
and to set the value to null when the Contributor is deleted
- adjust Facility get_contributors method to skip if
match.facility_list.contributor is None
- add `delete_contributor` Django management command that take an email
address as its argument and deletes the User, Contributor, and
EmailAddress objects associated with the email address

Connects #305 
Connects #269

## Demo

![Screen Shot 2019-03-18 at 5 36 31 PM](https://user-images.githubusercontent.com/4165523/54565932-e14f1a80-49a5-11e9-8d84-dff3f73f41ea.png)

->

![Screen Shot 2019-03-18 at 5 37 31 PM](https://user-images.githubusercontent.com/4165523/54565942-e7dd9200-49a5-11e9-9f7e-437b8902f6ce.png)

->

![Screen Shot 2019-03-18 at 5 37 12 PM](https://user-images.githubusercontent.com/4165523/54565953-f035cd00-49a5-11e9-8479-8305e8bfe74e.png)

-> (later, when added from another list)

![Screen Shot 2019-03-18 at 5 49 19 PM](https://user-images.githubusercontent.com/4165523/54566048-2d9a5a80-49a6-11e9-95af-78d8cb4d8a39.png)

## Testing Instructions

- get this branch, then `./scripts/resetdb`
- register for a new user through the UI, then upload list 11 and batch_process all steps to create some facities
- search for some facilities and verify that they appear and that your user/list is displayed among the contributors on its details page
- run `./scripts/manage delete_contributor --email <YOUR_EMAIL>` and verify that it succeeds
- visit the map page and search for facilities. Verify that that works and then visit the details page for a few facilities to observe that the now have no contributors
- register for a new user with a different email address, then upload list 11 again, process it, then look at facilities details pages and verify that the new user is listed among the contributors
- verify that you can delete a user through the Django admin
